core/src/replay_stage.rs

```diff
@@ -1697,6 +1701,41 @@ impl ReplayStage {
                     replay_vote_sender,
                     verify_recyclers,
                 );
+                debug!(
+                    "replayed into bank slot: {:?} / {:?}, result: {:?}, the execute_timings stats are: {:?}",
+                    bank.slot(),
+                    bank_slot,
+                    replay_result,
+                    bank_progress.replay_stats.execute_timings
+                );
+                let mut cost_model_mutable = cost_model.write().unwrap();
+                for (program_id, stats) in &bank_progress
```

Comment:
Does the cost model get pruned for programs which have not been executed in a long time?

Reply:
Good point, if one intended to attack by repeatedly sending transactions with random program keys, it is a security risk. 

One solution is to set cost table capacity upfront, once its capacity is reached, it will have to push out oldest record to make room to new one. With this, there is no need to add additional logic to persisting/restoring table.

Reply:
Add ExecuteCostTable with fixed capacity; when capacity is reached, adding new program data will trigger pruning. Pruning removes program that are old and less frequently executed.